src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -92,9 +92,18 @@ private int ParentProcessId
         }
 
         /// <summary>Gets the path to the current executable, or null if it could not be retrieved.</summary>
-        private static string GetExePath()
+        private static string? GetExePath()
         {
-            return Interop.libproc.proc_pidpath(Environment.ProcessId);
+            try
+            {
+                return Interop.libproc.proc_pidpath(Environment.ProcessId);
+            }
+            catch (Win32Exception)
+            {
+                // It will throw System.ComponentModel.Win32Exception (2): No such file or Directory when
```

Comment:
Should it filter for NativeErrorCode == 2?

Reply:
@danmosemsft Thank you. I agree with you.

Reply:
I see this as a temporary workaround until we get the proper API that returns the executable path in place. I do not think it is that important to make this condition perfect, and it is likely that the `proc_pidpath` has other failure modes e.g. permission errors.

Reply:
Sounds good